### ISSUE TITLE
Retrieve 'feature_importances` from canned BoostedTreesRegressor and BoostedTreesClassifier

### ISSUE THREAD

**Issue by dvdbisong** [tensorflow/tensorflow#21204]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: N/A
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: N/A
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: N/A
- **TensorFlow version (use command below)**: N/A
- **Python version**: N/A
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

### Feature request

When using the Premade BoostedTreesRegressor or Classifier in TensorFlow to train a model, it will be nice to have a way to retrieve the `feature_imporances` or feature ranking to know which variables have the most effect in the resulting model as seen in Scikit-learn's implementation.

Perhaps it can be something like, `trees.get_variable_value('feature_imporances')` to retrieve the feature rank in the order of the inputs.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by nataliaponomareva** [comment]
This is not supported as of now in a canned version. You can use the contrib version tensorflow/contrib/boosted trees, which can export feature importances (see boston.py example)

**Comment by martinwicke** [comment]
Should this be kept open as a feature request? Would you accept contributions?

**Comment by nataliaponomareva** [comment]
Sure, contributions are welcome. Technically all the data is already kept around - it is gain in NodeMetadata. One would need to go through all the nodes of all the trees and sum the gains groupped by the feature ids.

**Comment by facaiy** [comment]
I'm interested in the issue, and I'd like to take a try :-)

**Comment by martinwicke** [comment]
Go for it. Mention @nataliaponomareva for review when you have something. Thanks!

**Comment by facaiy** [comment]
Nice. I just forget to remind that PR #21509 is created to fix the issue.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.
